tenpy/algorithms/mps_sweeps.py

```diff
@@ -448,21 +475,18 @@ def combine_Heff(self):
         do we need both LP and RP or can we get away with just one? Is there a
         preference for one or the other?
         """
-        LHeff = npc.tensordot(self.LP, self.W, axes=['wR', 'wL'])
-        pipeL = LHeff.make_pipe(['vR*', 'p'])
-        self.LHeff = LHeff.combine_legs([['vR*', 'p'], ['vR', 'p*']],
-                                        pipes=[pipeL, pipeL.conj()],
-                                        new_axes=[0, -1])
-        RHeff = npc.tensordot(self.RP, self.W, axes=['wL', 'wR'])  #single-site.
-        pipeR = RHeff.make_pipe(['p', 'vL*'])
-        self.RHeff = RHeff.combine_legs([['p', 'vL*'], ['p*', 'vL']],
-                                        pipes=[pipeR, pipeR.conj()],
-                                        new_axes=[-1, 0])
-        self.pipeL = pipeL
-        self.pipeR = pipeR
-
-        self.pipesL = [pipeL, self.RP.make_pipe(['vL*'])]
-        self.pipesR = [pipeR, self.LP.make_pipe(['vR*'])]
+        if self.move_right:
```

Comment:
I seem to remember there was a reason for wanting both `RHeff` and `LHeff` to be available all the time.

Reply:
The second LHeff when moving left was needed before in the update_LP, but I added the extra if clause there.
The whole extra code in update_LP (beside the env.get_LP) is just optimization for `combine=True` to avoid one more tensordot.
Doing the tensordot here because we might need it later (but most of the times don't need it) - that would be an overkill of optimization.